### DISCUSSION TITLE
Global Metric Range Picker

### DISCUSSION THREAD

**Discussion by jdc1898** [laravel/nova-issues#6168] [Ideas & Feature Requests]
I would love to see a feature where you could “link” all metric cards Time Range filters together. This way on my dashboard page I could change the range from 30days to 90 days and all would update.

Maybe I could define a custom “Global” filter called ‘DashboardDatePicker’

When I choose an option in that picker, it updates all other metrics linked to it.

(new MetricOne)->refreshOnFilterChange()->linkTo(DashboardFilter)

**Comment by pratik149** [comment]
+1 Would really like to see this feature getting implemented!

**Comment by donparapidos** [comment]
+1 Any plans for delivering such feature ?? Currently having 20+ metrics with 20+ range selectors is really not useful and time consuming

**Comment by jeremynikolic** [comment]
Thanks for reaching to us with improvement ideas 🙏 
We are currently in the final preparation phase of the next major release and have "blocked" scope for i.
We will therefore resume looking at new improvements right after 👍

**Reply by jdc1898** [reply]
This is great news! Happy to help test it out!

**Comment by jeremynikolic** [comment]
I've given this some thoughts, some limitation here would be that all metrics needs to be using the same ranges otherwise the UX would be rather weird.
Do you agree this would be a rather obvious limitation on that feature ?

**Reply by donparapidos** [reply]
Yes, it makes no sense to have different range for each metric, take Grafana's implementation, or AWS's Cloudwatch for example. You have one global range selector that is applied to all metrics. That being said, and taking those big player's implementations, it is not limitation but expected behaviour.

**Comment by jdc1898** [comment]
I agree with this as well.

If you have multiple metrics on a page it would make sense that only one interval be active for any tagged metric. Maybe it could be something added to the metric itself?


Metric

/**
 *
 * Set’s if the metric should be part of the global date time filter.
 *
 */
 public function syncFilters()
{
return [
‘daily’,
‘weekly’,
// ‘monthly’
// Yearly'
];
}

If a Resource has a Card with a Metric returned where sync is true then the global filter option appears.

Maybe we could on the Resource Card have options too.

(new EmailDeliveryStats())->syncFilters(),
(new DomainStats())

Just some ideas maybe...


On Sep 19, 2024, at 2:19 AM, Goran Parapid ***@***.***> wrote:



Yes, it makes no sense to have different range for each metric, take Grafana's implementation, or AWS's Cloudwatch for example. You have one global range selector that is applied to all metrics. That being said, and taking those big player's implementations, it is not limitation but expected behaviour.
